src/Datasource/EntityTrait.php

```diff
@@ -353,6 +353,10 @@ public function patch(array $values, array $options = [])
      */
     protected function isModified(string $field, mixed $value): bool
     {
+        if (!array_key_exists($field, $this->_fields)) {
```

Comment:
You got to check for null on fields first here

Reply:
I can't figure it out. Where should it be moved?

Reply:
```suggestion
        if ($this->_fields === null || !array_key_exists($field, $this->_fields)) {
```

Reply:
At least i assume given the failed tests